### PR TITLE
update docs for the upcoming release, add changelog

### PR DESCRIPTION
**Pull request by gabrielittner**
- merge the codegen navigation section into the main codegen docs since standalone isn't supported anymore
- add docs for the new Activity codegen

Note: this already includes the changes from #545

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,9 +1,44 @@
...
+- `com.freeletics.khonshu:navigation` is now a multiplatform module with JVM and Android
+  targets. The goal is not to support multiplatform navigation (for now at least), but to
+  make it possible to reference `NavRoute` in Kotlin/JVM modules.
+
+
+## Codegen
+
+- **Breaking**: Renamed the Compose `@ComposeDestnation` annotation to `@NavDestination` and
+  the Fragment `@ComposeDestination` to `@ComposeFragmentDestination`.
+- **Breaking**: Removed `@ScopeTo` and `ForScope` annotations. They have been replaced by
```
Comment (kboyarshinov) [inline CHANGELOG.md:20]:
```suggestion
- **Breaking**: Removed `@ScopeTo` and `@ForScope` annotations. They have been replaced by
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,9 +1,44 @@
...
+  Anvil's `@SingleIn` and `@ForScope` from `com.squareup.anvil:annotations-optional`.
+- New: The code generation now supports KSP. The functionality is generally the same
+- New: A new `@NavHostActivity` annotation was added and will generate an `Activity` and the
```
Comment (kboyarshinov) [inline CHANGELOG.md:23]:
```suggestion
- **New**: The code generation now supports KSP. The functionality is generally the same
- **New**: A new `@NavHostActivity` annotation was added and will generate an `Activity` and the
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,9 +1,44 @@
...
+  related boilerplate. Check out [the docs](https://freeletics.github.io/khonshu/codegen/) for
+  more details.
+- It's now possible to use both `@NavDestination` and `@ComposeFragmentDestination` at the same
+  time on the same composable. This allows transitioning from Fragments to Compose more easily.
+- Removed: `@RendererScreen` and `@ComposeScreen` annotations. These allowed generating Fragments
```
Comment (kboyarshinov) [inline CHANGELOG.md:28]:
```suggestion
- **Removed**: `@RendererScreen` and `@ComposeScreen` annotations. These allowed generating Fragments
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,9 +1,44 @@
...
+  and Composables without relying on Khonshu navigation. We've only ever used this for Activities
+  which now have a better solution. In the beginning we saw the codegen and navigation as 2
+  separate things but now the codegen is more of an add-on, so we decided to remove the standalone
+  mode for codegen.
```
Comment (kboyarshinov) [inline CHANGELOG.md:32]: By standalone do you mean "without navigation", right?

File: docs/codegen/activity-generation.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+```kotlin
+ // a scope marker used to represent the `Activity`.
+sealed interface ExampleActivityScope
+
+@NavHostActivity(
+    scope = ExampleActivityScope::class,
+    parentScope = AppScope::class, // AppScope is the default value and can be omitted
+    stateMachine = ExampleStateMachine::class,
+    activityBaseClass = ComponentActivity::class, // the super class for the generated activity
```
Comment (kboyarshinov) [inline docs/codegen/activity-generation.md:16]:
```suggestion
    activityBaseClass = ComponentActivity::class, // the super class for the generated activity, ComponentActivity is the default value and can be omitted
```

**Review by kboyarshinov — APPROVED** [review]
Left couple of suggestions but looks good 👍

File: CHANGELOG.md (hunk 4 above)
Comment (gabrielittner) [reply CHANGELOG.md:32]: Yes
```suggestion
  separate things but now the codegen is more of an add-on, so we decided to remove the standalone
  mode (codegen without navigation) for codegen.
```

File: docs/codegen/activity-generation.md (hunk 5 above)
Comment (gabrielittner) [reply docs/codegen/activity-generation.md:16]: This one doesn't have a default value right now. Partially because it lives in the common source set.

**Pull request merged**